### PR TITLE
feat(worker/aave)

### LINKED ISSUE #94: aave
### Tag

Exchange

### Project URL

https://aave.com/

### Worker description

aave worker

### What actions should be included?

1. Deposit stETH to AAVE V2 LendingPool on Ethereum Mainnet
2. Withdraw USDC from AAVE V2 LendingPool on Ethereum Mainnet
3. Borrow USDT from AAVE V2 LendingPool on Ethereum Mainnet
4. Repay USDC to AAVE V2 LendingPool on Ethereum Mainnet
5. Supply rETH to AAVE V3 Pool on Ethereum Mainnet
6. Supply USDCbC to AAVE V3 Pool on Ethereum Base
7. Withdraw WBTC from AAVE V3 Pool on Ethereum Mainnet
8. Borrow USDT from AAVE V3 Pool on Ethereum Mainnet
9. Repay USDT to AAVE V3 Pool on Ethereum Mainnet

### Additional description

_No response_

### This is not a duplicated worker request

- [X] I have searched [existing issues](https://github.com/NaturalSelectionLabs/RSS3-Node/issues) and [pull requests](https://github.com/NaturalSelectionLabs/RSS3-Node/pulls) to ensure this worker has not already been requested

### PR DESCRIPTION
**Pull request by Candinya**
## Checklist

- [x] The commit message follows [Angular Contributing guidelines](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit);
- [x] Tests for the changes have been added (for bug fixes / features);

## Project

AAVE

### URLs

- https://aave.com/

### Networks

- Mainnet
- Polygon
- Avalanche
- Base
- Optimism
- Arbitrum
- Fantom

### Contract addresses

- AddressV1LendingPool `0x398eC7346DcD622eDc5ae82352F02bE94C62d119`
- AddressV2LendingPool
  - Mainnet `0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9`
  - Polygon `0x8dFf5E27EA6b7AC08EbFdf9eB090F32ee9a30fcf`
  - Avalanche `0x4F01AeD16D97E3aB5ab2B501154DC9bb0F1A5A2C`
- AddressV3Pool
  - Mainnet `0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2`
  - Base `0xA238Dd80C259a72e81d7e4664a9801593F98d1c5`
  - Others `0x794a61358D6845594F94dc1DB02A252b5b4814aD`

### Transactions

- Mainnet
  - Deposit stETH to AAVE V2 LendingPool on Ethereum Mainnet `0x488591689796c234eb6a876be238c67a9092d755070d6992dddccd2f8beee57c`
  - Withdraw USDC from AAVE V2 LendingPool on Ethereum Mainnet `0x759532f09bd943d6da2964d58d7fc9a6b3f3914ef0643c37cb5f0eb1f3fe03a6`
  - Borrow USDT from AAVE V2 LendingPool on Ethereum Mainnet `0xd01b5eab0d6abceb5f21da3dc6c329b35b5f854fd4c0e7d05b2cd81e9a410c9f`
  - Repay USDC to AAVE V2 LendingPool on Ethereum Mainnet `0x3c309fa16fb7e995fba8f236d2eb3a20a44f3ce116a13899683f4580154e85ec`
  - Supply rETH to AAVE V3 Pool on Ethereum Mainnet `0x2626b06195b4e7699495b1ea542a954a7b164195f3fa2dbd1c063882cf697bce`
  - Withdraw WBTC from AAVE V3 Pool on Ethereum Mainnet `0xef6e2adee0defc810147847b2c3c5255e5e5e75cbc7a4e6ff009444f0aa53f11`
  - Borrow USDT from AAVE V3 Pool on Ethereum Mainnet `0x060a2a7e1db5be295f14171e98a39dec1a7282101c148833333409b99dad4295`
  - Repay USDT to AAVE V3 Pool on Ethereum Mainnet `0xdc2cd838127a068db79c70508ac87b73d083a2ea22d230ae6f1d8b27001e58d0`
- Base
  - Supply USDCbC to AAVE V3 Pool on Ethereum Base `0x3ee309226dfd6a4d26be87082e5e1ce6c8769f130c9495b2e44684ad5cc2682a`

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

resolve #94

### CODE REVIEW

File: deploy/prod/values.yaml [hunk 1]
```diff
@@ -64,6 +64,41 @@ indexers:
     parameters:
       block_height_start: 1275799
       rpc_thread_blocks: 1
+  - network: ethereum
```
Comment (kallydev) [inline deploy/prod/values.yaml:67]: We also need to add the secrets to Vault.

File: provider/ethereum/contract/multicall3/contract.go [hunk 2]
```diff
@@ -24,7 +24,11 @@ var (
 var deployedAtMap = map[uint64]uint64{
 	uint64(filter.EthereumChainIDMainnet):  14353601, // https://etherscan.io/tx/0x00d9fcb7848f6f6b0aae4fb709c133d69262b902156c85a473ef23faa60760bd
 	uint64(filter.EthereumChainIDOptimism): 4286263,  // https://optimistic.etherscan.io/tx/0xb62f9191a2cf399c0d2afd33f5b8baf7c6b52af6dd2386e44121b1bab91b80e5
-	uint64(filter.EthereumChainIDPolygon):  15921452, // https://bscscan.com/tx/0xcc0ddf5f791617ba9befce57995dbcb3a202946a1eefa3469742b01a0decdaf2
+	//uint64(filter.EthereumChainIDBSC):      15921452, // https://bscscan.com/tx/0xcc0ddf5f791617ba9befce57995dbcb3a202946a1eefa3469742b01a0decdaf2
```
Comment (kallydev) [inline provider/ethereum/contract/multicall3/contract.go:27]: Is this code not used?

**Review by kallydev — APPROVED** [review]

File: provider/ethereum/contract/multicall3/contract.go (hunk 2 above)
Comment (Candinya) [reply provider/ethereum/contract/multicall3/contract.go:27]: BSC chain id not found in filter package, the original code seems like a typo to polygon, support for BSC (w/ RPCs etc) might need be marked as a TODO item

File: deploy/prod/values.yaml (hunk 1 above)
Comment (Candinya) [reply deploy/prod/values.yaml:67]: Yes, RPCs to add:

- AVALANCHE
- BASE
- OPTIMISM
- ARBITRUM
- FANTOM

(seems like ETHEREUM and POLYGON already set)

Comment (pseudoyu) [reply deploy/prod/values.yaml:67]: i'll add extra rpcs in vault, highlight also uses these blockchain

Comment (pseudoyu) [reply deploy/prod/values.yaml:67]: All set.

**Review by pseudoyu — DISMISSED** [review]

**Review by pseudoyu — APPROVED** [review]

**Pull request merged**